packages/astro/src/integration/create-integration.ts

```diff
@@ -110,7 +110,16 @@ function createIntegration<Params extends HotloadAstroClerkIntegrationParams>()
             `
             ${command === 'dev' ? `console.log("${packageName}","Initialize Clerk: page")` : ''}
             import { runInjectionScript } from "${buildImportPath}";
-            await runInjectionScript(${JSON.stringify(internalParams)});`,
+
+            await runInjectionScript(${JSON.stringify(internalParams)});
+
+            document.addEventListener('astro:before-swap', async e => {
+              window.Clerk.__internal_flushEmotionCache();
+            });
```

Comment:
This will run before the new document replaces the current document. See more info [here](https://docs.astro.build/en/guides/view-transitions/#astrobefore-swap)